### PR TITLE
Post link redirect to the page with the post.

### PR DESCRIPTION
**Pull request by koca2000**
This PR makes a redirect during post loading if the post the user wanted to see is on another page. This is most commonly used in `recentposts` hcm module where user clicks on a link of a specific post.

### CODE REVIEW

File: system/class/Post/PostService.php [hunk 1]
```diff
@@ -297,6 +297,28 @@ static function renderList(int $style, int $home, $vars, bool $force_locked = fa
         /* --- init pager --- */
         $paging = Paginator::render($url, $postsperpage, DB::table('post'), $countcond, "#posts", $page_param, $autolast);
 
+        /* --- redirect to page with post from url --- */
+        $selected_post_id = Request::get("post");
+        if ($selected_post_id !== null) {//" . $pageCond . " AND 
+            $selected_post = DB::queryRow("SELECT p.id, p.xhome FROM " . DB::table('post') . " p WHERE p.id = " . DB::esc($selected_post_id));
+            
+            if ($selected_post !== false){
```
Comment (jDanek) [inline system/class/Post/PostService.php:305]: Missing space between brackets `){`

File: system/class/Post/PostService.php [hunk 2]
```diff
@@ -297,6 +297,28 @@ static function renderList(int $style, int $home, $vars, bool $force_locked = fa
...
+                $post_id = $selected_post_id;
+                if ($style !== self::RENDER_FORUM_TOPIC && $style !== self::RENDER_PM_LIST && $selected_post['xhome'] != -1)
```
Comment (jDanek) [inline system/class/Post/PostService.php:307]: If statements without brackets

File: system/class/Post/PostService.php [hunk 3]
```diff
@@ -297,6 +297,28 @@ static function renderList(int $style, int $home, $vars, bool $force_locked = fa
...
+                    $post_id =  $selected_post['xhome'];
```
Comment (jDanek) [inline system/class/Post/PostService.php:308]: Two spaces after the equals sign

File: system/class/Post/PostService.php [hunk 4]
```diff
@@ -297,6 +297,28 @@ static function renderList(int $style, int $home, $vars, bool $force_locked = fa
...
+                
+                $post_number = DB::count('post', $countcond . ' AND id ' . ($desc !== '' ? '>= ' : '<= ') . DB::esc($post_id) . ' ORDER BY ' . ($is_topic_list ? 'p.sticky DESC,' : '') . $ordercol . ' ' . $desc); 
+
+                $post_page = max(1, ceil($post_number / $postsperpage));
+                if ($post_page != $paging['current']){
```
Comment (jDanek) [inline system/class/Post/PostService.php:313]: Missing space between brackets `){`

**Review by jDanek — COMMENTED** [review]
Code style is fine by me, but I haven't tested the functionality.

**Comment by ShiraNai7** [comment]
This was never handled probably because it was assumed that "recent" posts would still be on the first page.

I'm not sure that redirecting from `PostService::renderList()` is the right place to do it. I'll look into this - this logic is already copy-pasted in multiple places and could be reused for this as well.

**Comment by koca2000** [comment]
> This was never handled probably because it was assumed that "recent" posts would still be on the first page.

That is logical assumption but it doesn't work for forums since the newest post is usually on the last page and not the first page.

> I'm not sure that redirecting from `PostService::renderList()` is the right place to do it. I'll look into this - this logic is already copy-pasted in multiple places and could be reused for this as well.

I agree. It is not the perfect place. I chose it because it is the place where pagination parameters are known and also it would be applied for all post types.

**Comment by ShiraNai7** [comment]
This should be fixed in cc0b9cf245db48b027a84bca36f998210aea757e

**Pull request closed without merging**